subprojects/docs/src/samples/build-organization/composite-builds/plugin-dev/kotlin/greeting-plugin/src/main/java/org/sample/GreetingTask.java

```diff
@@ -1,23 +1,20 @@
 package org.sample;
 
 import org.gradle.api.DefaultTask;
+import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.TaskAction;
 
-public class GreetingTask extends DefaultTask {
-    private String who = "mate";
-
-    @Input
-    public String getWho() {
-        return who;
+public abstract class GreetingTask extends DefaultTask {
+    public GreetingTask() {
+        getWho().convention("mate");
```

Comment:
I think it might be more proper to configure this in the extension. I know it wasn't before, but it's another bad example.

Reply:
Good call